### ISSUE TITLE
Add support for 'use_index' property on pouchdb-find queries

### ISSUE THREAD

**Issue by virtuakazib** [pouchdb/pouchdb#6452]
### Issue
pouchdb-find currently attempts to find a suitable query index for each query made.  But the logic by which it selects one index over another isn't always clear.  And sometimes it selects an unexpected index which causes unexpected query results.  The `_find` endpoint on both CouchDB and Cloudant support a 'use_index' property on queries that allows the client to explicitly name a previously created index that should be used with the query.

pouchdb-find already allows indexes to be named when they are created.  Queries should be able to use those names to specify which index to use.

#### Format
```
"use_index": "<design_document>"
```
or
```
"use_index: ["<design_document>", "<index_name>"]
```

#### Usage
```
db.find({
  selector: {
    name: {$gte: null}
  },
  sort: ['name'],
  limit: 10,
  use_index: ['designDoc', 'indexName']
});
```

### Info
- Adapter: pouchdb-find

**Comment by faizanahemad** [comment]
+1

**Comment by garrensmith** [comment]
Fixed in #6497
